rollup-common.js

```diff
@@ -40,54 +40,75 @@ const skipBundleOutput = {
   },
 };
 
-const reservedProperties = [
-  '_$litType$',
-  '_$litDirective$',
-  // TODO Decide on public API
-  // https://github.com/Polymer/lit-html/issues/1261
-  '_value',
-  '_setValue',
-  'createTreeWalker',
-  // Note, used in platform-support and reserved so platform-support can work
-  // x-version.
-  '_handlesPrepareStyles',
-  // TODO(kschaaf) TBD: lit-ssr required "private" fields (can be in
-  // crossPackagePropertyMangles once lit-ssr uses the rollup config)
-  // lit-html: AttributePart (used by render-lit-html)
-  '_commitValue',
-  // lit-html: Directive (used by render-lit-html)
-  '_resolve',
-  // updating-element: UpdatingElement (used by lit-element-renderer)
-  '_attributeToProperty',
-  // hydrate-support: LitElement (added by hydrate-support)
-  '_needsHydration',
-];
+// Private properties which should never be mangled. They need to be long/obtuse
+// to avoid collisions since they are used to brand values in positions that
+// accept any value. We don't use a Symbol for these to support mixing and
+// matching values from different versions.
+const reservedProperties = ['_$litType$', '_$litDirective$'];
 
-// Any private properties which we share between different _packages_ are
-// hard-coded here because they must never change between versions. Mangled
-// names are uppercase letters, in case we ever might want to
-// use lowercase letters for short, public APIs.
-// Note, these are used for `platform-support`.
-const crossPackagePropertyMangles = {
-  // lit-html: Template
-  _createElement: 'A',
-  _element: 'B',
-  _options: 'C',
-  // lit-html: NodePart
-  _startNode: 'D',
-  _endNode: 'E',
-  _getTemplate: 'F',
-  // lit-html: TemplateInstance
-  _template: 'H',
-  // updating-element: UpdatingElement
-  _didUpdate: 'S',
-  _controllers: 'T',
+// Private properties which should be stable between versions but are used on
+// unambiguous objects and thus are safe to mangle. These include properties on
+// objects accessed between packages or objects used as values which may be
+// accessed between different versions of a given package.
+//
+// By convention, stable properties should be prefixed with `_$` in the code so
+// they are easily identifiable as properties requiring version stability and
+// thus special attention.
+//
+// Mangled names are uppercase letters, in case we ever might want to use
```

Comment:
I think we should consider prefixing with `_` for names of properties that end up on publicly accessible objects, like Parts and TemplateResults.

Reply:
This is just a list of the "special" mangled properties that need to be stable; there will be other mangled properties on e.g. Parts that will be whatever Terser picks for it. Does this suggest that all Terser mangling should use `_`?

Reply:
oh right... we can't control the naming for the other properties. nvm then. If terser ever allows this we can enable it broadly. The trick would be to only incur the extra char on properties of publicly accessible objects.